### PR TITLE
Patch 1

### PR DESCRIPTION
**Pull request by clin1234**

### CODE REVIEW

**Comment by hieunc229** [comment]
Charlie, I don't think using `indexOf` with an array of strings will return the desired index value. 

Screen Shot 2021-07-18 at 9 48 53 am

Though it's a real issue. Some people use the current language comment annotation instead of double forward slash `//`.

There is another approach to this. The extension is registered using glob pattern, I think we can update the pattern
https://github.com/hieunc229/copilot-clone/blob/master/src/extension.ts#L58

**Comment by clin1234** [comment]
Suggestions on handling HTML and XML-style comments?

File: src/extension.ts [hunk 1]
```diff
@@ -22,7 +22,7 @@ export function activate(context: vscode.ExtensionContext) {
 				new vscode.Range(position.with(undefined, 0), position)
 			);
 
-			if (textBeforeCursor.indexOf(CSConfig.SEARCH_PHRASE_START) != -1 && textBeforeCursor[textBeforeCursor.length - 1] === CSConfig.SEARCH_PHRASE_END) {
+			if (CSConfig.SEARCH_PHRASE_START.includes(textBeforeCursor) && textBeforeCursor[textBeforeCursor.length - 1] === CSConfig.SEARCH_PHRASE_END) {
```
Comment (hieunc229) [inline src/extension.ts:25]: I don't think `includes` works either. 

In JS, the `includes` function is checking if the "whole" value of`textBeforeCursor` exists in the `SEARCH_PHRASE_START` string array, not just part of `textBeforeCursor`

Comment (Kek5chen) [reply src/extension.ts:25]: I think what you two want would be

`textBeforeCursor.includes(CSConfig.SEARCH_PHRASE_START)`

The includes method has been overwritten in a string object, it doesnt act like in an array. In the context of a string it is used to see if a string contains something.

```
const string = "foo";
const substring = "oo";

console.log(string.includes(substring));
```

Console Output:
`true`

Comment (Kek5chen) [reply src/extension.ts:25]: This is still unrelevant though as we are changing to regex at this point and therefore this becomes irrelevant.

**Comment by Kek5chen** [comment]
This pull request is irrelevant as pull request #16 is an upgraded and better version of the exact same concept.

**Pull request closed without merging**